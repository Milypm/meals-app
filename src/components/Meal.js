import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { Link } from 'react-router-dom';
import { passDetails, updateDetailsStatus } from '../actions/index';

const Meal = (props) => {
  const dispatch = useDispatch();
  const { name, img, id } = props;
  const dispatches = () => {
    dispatch(passDetails(id));
    dispatch(updateDetailsStatus(true));
  };
  return (
    <Link to="/recipe-details" className="mealObj" onClick={() => dispatches()}>
      <h3 className="meal-title">{name}</h3>
      <img src={img} alt={name} />
    </Link>
  );
};
Meal.propTypes = {
  name: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
};
export default Meal;
