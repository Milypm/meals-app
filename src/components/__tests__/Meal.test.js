import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
import Meal from '../Meal';

describe('Meal component', () => {
  test('Meal component renders correctly', () => {
    const tree = renderer.create(
      <BrowserRouter>
        <Meal
          key="53014"
          id="53014"
          name="Pizza Express Margherita"
          img="https://www.themealdb.com/images/media/meals/x0lk931587671540.jpg"
        />
      </BrowserRouter>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
