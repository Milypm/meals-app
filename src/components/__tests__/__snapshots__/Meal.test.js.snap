// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meal component Meal component renders correctly 1`] = `
<a
  className="mealObj"
  href="/recipe-details/53014"
  onClick={[Function]}
>
  <h3
    className="meal-title"
  >
    Pizza Express Margherita
  </h3>
  <img
    alt="Pizza Express Margherita"
    src="https://www.themealdb.com/images/media/meals/x0lk931587671540.jpg"
  />
</a>
`;
